src/memo/memo.ml

```diff
@@ -763,17 +786,17 @@ end = struct
         (T { without_state = dep_node.without_state; running_state })
         (fun () ->
           match dep_node.without_state.spec.f with
-          | Function.Async f -> f inp)
+          | Function.Async f -> Fiber.collect_errors (fun () -> f inp))
```

Comment:
I just realised that here is another example of poor error-handling API: there is no way to get rid of the impossible case with the empty list of collected exceptions.